charts/redpanda/client/client.go

```diff
@@ -415,3 +434,68 @@ func wrapTLSDialer(dialer DialContextFunc, config *tls.Config) DialContextFunc {
 		return tls.Client(conn, config), nil
 	}
 }
+
+// srvLookup performs an SRV DNS lookup on the given helm release as a form of service discovery.
+//
+// As with all forms of service discovery, this method may miss Pods that are
+// temporarily unavailable at the time of invocation.
+//
+// If dialer is nil, this function assumes that it's being executed from within
+// a Kubernetes and performs a DNS query through the default resolver.
+//
+// See also: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#srv-records
+func srvLookup(dot *helmette.Dot, dialer DialContextFunc, service string) ([]*net.SRV, error) {
+	// To preserve backwards compatibility of the top level client
+	// constructor's methods, we use a context with a static timeout.
+	// While less than ideal, 30s should be a reasonable upper limit for this method.
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// A nil / zero resolver is valid. In the case that dialer is nil, we
+	// assume our DNS requests will go to kube-dns and appropriately resolve.
+	var resolver net.Resolver
+
+	if dialer != nil {
+		// Otherwise, we need to do a bit of leg work to perform an in-cluster
+		// DNS query from outside of the cluster.
+		ctl, err := kube.FromRESTConfig(dot.KubeConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		resolver = net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
```

Comment:
😬 -- this seems dirty, but I understand why it's here... though I don't necessarily like the fact that dialer now == manually resolving and dialing the DNS pod and assuming we are outside of the kubernetes network. Can you change the parameter names for `network`, and `address` to start with underscores just to show they're not used?

Reply:
We could probably do some other forms of introspection (e.g. of the kubeconfig) to determine if we're in a cluster or not. For now I'm going to go with the build tag as that seems safest.